internal/exec/util/file.go

```diff
@@ -136,7 +136,8 @@ func (u Util) WriteLink(s types.Link) error {
 	}
 
 	if s.Hard {
-		return os.Link(s.Target, path)
+		targetPath := u.JoinPath(s.Target)
```

Comment:
Is this the test in question? https://github.com/arithx/ignition/blob/12f1c211142ee20cfd7e098a763ff033cb9a8f10/tests/test_generator.go#L544